javaparser-core/src/main/java/com/github/javaparser/resolution/types/ResolvedReferenceType.java

```diff
@@ -471,6 +470,10 @@ protected boolean compareConsideringTypeParameters(ResolvedReferenceType other)
             for (int i = 0; i < typeParametersValues().size(); i++) {
                 ResolvedType thisParam = typeParametersValues().get(i);
                 ResolvedType otherParam = other.typeParametersValues().get(i);
+                //Sometimes, thisParam and otherParam may be LazyType.
```

Comment:
nitpick: I would use a space after the slash and a space between the if and the parenthesis

Reply:
Perhaps that means that two types can be different but have the same description?

Reply:
Hi, @ftomassetti, thanks for the advice. I will be careful about the code format. 

I actually want to compare the qualified name of two types and I know the `describe()` method doesn't return the class name. I can call `getQualifiedName()` method on `ResolvedReferenceType` object, however in `LazyType`, there is only a `describe()` method.

How about the following patch? This is not a perfect way, but I think this is more robust than `describe()` method.
```
if (thisParam.isReferenceType() && otherParam.isReferenceType() 
    && thisParam.asReferenceType().getQualifiedName().equals(otherParam.asReferenceType().getQualifiedName()))
    return true;
```


I tried to import `LazyType` in this `ResolvedReferenceType` class, however, this seems failed because of the dependency between Javaparser-core and Javaparser-symbolsolver-core.
If we can access the `LazyType` in this method, we can get the concrete type from the `LazyType`.

Reply:
This seems reasonable, let's try that